pkg/api/pluginproxy/pluginproxy.go

```diff
@@ -38,6 +39,29 @@ func getHeaders(route *plugins.AppPluginRoute, orgId int64, appID string) (http.
 	return result, err
 }
 
+func UpdateURL(route *plugins.AppPluginRoute, orgId int64, appID string) error {
+	query := m.GetPluginSettingByIdQuery{OrgId: orgId, PluginId: appID}
+	if err := bus.Dispatch(&query); err != nil {
+		return err
+	}
+
+	data := templateData{
+		JsonData:       query.Result.JsonData,
+		SecureJsonData: query.Result.SecureJsonData.Decrypt(),
+	}
+	interpolated, err := InterpolateString(route.Url, data)
+	if err != nil {
+		return err
+	}
+	// update the url of the route
+	route.Url = interpolated
```

Comment:
Wont this modify the route url permanently for all, is this instance of a AppPluginRoute not the source defintiion object?

Reply:
yes, i've fixed it and tested now, we're closing to be able to merge this (will remove the debug/trace info)